stablehlo/dialect/Base.h

```diff
@@ -78,6 +79,47 @@ class BoundedDialectInterface
   virtual Attribute createBoundedAttr(ArrayRef<int64_t> bounds) const = 0;
 };
 
+// This interface is used for HLO dialect bytecoding. It provides utility
+// methods for StableHLO and CHLO bytecode.
+class BaseBytecodeDialectInterface : public BytecodeDialectInterface {
```

Comment:
Do `readEnumAttribute` and `writeEnumAttribute` have to be inside a subclass of `BytecodeDialectInterface` to work? If not, I propose we don't introduce `BaseBytecodeDialectInterface` and turn these into free functions.

Reply:
They don't have to be. I thought the grouping may be nice, instead of base being a collection of free-floating functions. But if that's the intention, free works. Will fix!